apps/studio/src/pages/sites/[siteId]/folders/[folderId]/index.tsx

```diff
@@ -109,23 +143,205 @@ const FolderPage: NextPageWithLayout = () => {
           </HStack>
         </VStack>
         <Box width="100%">
-          <ResourceTable siteId={siteId} resourceId={folderId} />
+          <ResourceTable
+            siteId={parseInt(siteId)}
+            resourceId={parseInt(folderId)}
+          />
         </Box>
       </VStack>
       <CreatePageModal
         isOpen={isPageCreateModalOpen}
         onClose={onPageCreateModalClose}
-        siteId={siteId}
+        siteId={parseInt(siteId)}
       />
       <CreateFolderModal
         isOpen={isFolderCreateModalOpen}
         onClose={onFolderCreateModalClose}
+        siteId={parseInt(siteId)}
+      />
+      <FolderSettingsModal
+        isOpen={isFolderSettingsModalOpen}
+        onClose={onFolderSettingsModalClose}
         siteId={siteId}
-        key={uniqueId()}
+        resourceId={parseInt(folderId)}
       />
     </>
   )
 }
 
 FolderPage.getLayout = AdminCmsSidebarLayout
 export default FolderPage
+
+interface FolderSettingsModalProps {
+  isOpen: boolean
+  onClose: () => void
+  siteId: string
+  resourceId: number
+}
+const FolderSettingsModal = ({
+  isOpen,
+  onClose,
+  siteId,
+  resourceId,
+}: FolderSettingsModalProps) => {
+  return (
+    <Modal isOpen={isOpen} onClose={onClose}>
+      <ModalOverlay />
+      <SuspendableModalContent
+        isOpen={isOpen}
+        siteId={siteId}
+        resourceId={resourceId}
+        onClose={onClose}
+      />
+    </Modal>
+  )
+}
+
+const SuspendableModalContent = ({
+  isOpen,
+  onClose,
+  siteId,
+  resourceId,
+}: FolderSettingsModalProps) => {
+  const { setValue, register, handleSubmit, watch, formState, getFieldState } =
+    useZodForm({
+      defaultValues: {
+        title: "",
+        permalink: "",
+      },
+      schema: editFolderSchema.omit({ siteId: true, resourceId: true }),
+    })
+  const [{ title: originalTitle, permalink: originalPermalink, parentId }] =
+    trpc.folder.readFolder.useSuspenseQuery({
+      siteId: parseInt(siteId),
+      resourceId,
+    })
```

Comment:
since we have the original title and permalink, shouldn't we fetch the data first before setting the default values? 

```suggestion
  const [{ title: originalTitle, permalink: originalPermalink, parentId }] =
    trpc.folder.readFolder.useSuspenseQuery({
      siteId: parseInt(siteId),
      resourceId,
    })
    
  const { setValue, register, handleSubmit, watch, formState, getFieldState } =
    useZodForm({
      defaultValues: {
        title: originalTitle,
        permalink: originalPermalink,
      },
      schema: editFolderSchema.omit({ siteId: true, resourceId: true }),
    })
```

Reply:
thanks for the catch - updated!